src/shared_modules/utils/zlibHelper.hpp

```diff
@@ -88,6 +97,126 @@ namespace Utils
             } while (len == sizeof(buf));
             outputFile.close();
         }
+
+        /**
+         * @brief Uncompress ZIP file and returns a list with the decompressed files.
+         *
+         * @param zipFilePath Compressed (.zip) file path.
+         * @param outputDir Folder where the output files will be stored.
+         * @return std::vector<std::string> List of decompressed files.
+         */
+        static std::vector<std::string> zipDecompress(const std::filesystem::path& zipFilePath,
+                                                      const std::filesystem::path& outputDir)
+        {
+            // Open .zip file.
+            auto unzFile {unzOpen(zipFilePath.c_str())};
+            if (!unzFile)
+            {
+                throw std::runtime_error {"Unable to open compressed file: " + zipFilePath.string()};
+            }
+
+            // Get .zip file information (amount of files, i.e.).
+            unz_global_info globalInfo;
+            if (unzGetGlobalInfo(unzFile, &globalInfo) != UNZ_OK)
+            {
+                unzClose(unzFile);
```

Comment:
These calls could be avoided by using a custom remover like this:

```c++
struct UnzFileDeleter {
    void operator()(unzFile file) const {
        unzClose(file);
    }
};

using UnzFilePtr = std::unique_ptr<void, UnzFileDeleter>;
```
```c++

// Open .zip file.
            UnzFilePtr unzFile {unzOpen(zipFilePath.c_str())};

```

[Look](https://github.com/wazuh/wazuh/blob/dev-14153-vulndet-refactor/src/shared_modules/utils/zlibHelper.hpp#L21) at the usage in `gzipDecompress`

Reply:
Great! Done.